src/lib.rs

```diff
@@ -498,6 +498,15 @@ impl<'a> Read for &'a [u8] {
     }
 }
 
+impl<'a> Read for &'a mut [u8] {
+    type ReadError = Void;
+
+    fn read(&mut self, buf: &mut [u8]) -> Result<usize, Self::ReadError> {
+        let mut immutable : &[u8] = self;
+        immutable.read(buf)
+    }
```

Comment:
This doesn't update `self` to point to unread part afterwards.

Reply:
Indeed, thank for pointing that out. I was really stumped as to fix this properly, came up with `core::mem::replace` trick - please have a look if that looks reasonable.

Reply:
Yep, that's the usual way to solve this.